controllers/object_controls.go

```diff
@@ -1285,24 +1313,15 @@ func TransformToolkit(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, n
 		} else if runtime == gpuv1.Docker.String() {
 			socketEnvvarName = "DOCKER_SOCKET"
 		}
-		setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), socketEnvvarName, DefaultRuntimeSocketTargetDir+sourceSocketFileName)
+		setContainerEnv(&(obj.Spec.Template.Spec.Containers[mainContainerIndex]), socketEnvvarName, DefaultRuntimeSocketTargetDir+sourceSocketFileName)
 
 		volMountSocketName := fmt.Sprintf("%s-socket", runtime)
 		volMountSocket := corev1.VolumeMount{Name: volMountSocketName, MountPath: DefaultRuntimeSocketTargetDir}
-		obj.Spec.Template.Spec.Containers[0].VolumeMounts = append(obj.Spec.Template.Spec.Containers[0].VolumeMounts, volMountSocket)
+		obj.Spec.Template.Spec.Containers[mainContainerIndex].VolumeMounts = append(obj.Spec.Template.Spec.Containers[mainContainerIndex].VolumeMounts, volMountSocket)
```

Comment:
minor improvement tip: Since we reference `obj.Spec.Template.Spec.Containers[mainContainerIndex]` multiple times, it will be nice to move it a variable and reference that variable everywhere

Reply:
Fixed